lamindb/_record.py

```diff
@@ -32,24 +34,33 @@
 
 
 def serialize(
-    data: Union[Path, str, pd.DataFrame, ad.AnnData], name, format
-) -> Tuple[Any, Path, str, str]:
+    data: Union[Path, CloudPath, str, pd.DataFrame, ad.AnnData], name, format
+) -> Tuple[Any, Union[Path, CloudPath], str, str]:
     """Serialize a data object that's provided as file or in memory."""
     memory_rep = None
-    if isinstance(data, (Path, str)):
-        local_filepath = Path(data)
-        name, suffix = get_name_suffix_from_filepath(local_filepath)
+    if isinstance(data, str):
+        try:
+            data = CloudPath(data)
```

Comment:
I think this should error here if the returned cloudpath object doesn't equate the one in settings

Hence, e.g., the following code block
```
from lndb_setup import settings
if data != settings.instance.storage.root:
    raise ValueError(
        "Can only track objects in configured cloud storage locations. Please call `lndb_setup.set_storage("<bucket_name>")`.")
```

Reply:
Done!